File: app/src/main/java/com/br/amber/adedonha/models/Room.kt
package com.br.amber.adedonha.models

data class Room(
    val numberOfRounds: Int,
    var currentRound: Int = 0,
    val themes: List<String>,
    var drawnLetters: MutableList<String>,
    var stop: Boolean = false,
    var player1: Player,
    var player2: Player,
    var partnerFound: Boolean = false
) {

    constructor(): this(0, 0, listOf(), mutableListOf(),false, Player(), Player())


    companion object{
        fun createMutableListOfLettersWithNullValues(size: Int): MutableList<String>{
            val mutableListWithEmptyString = mutableListOf<String>()
            for (i in 1..size){
                mutableListWithEmptyString.add("")
            }
            return mutableListWithEmptyString
        }

    }



}

File: src/app/models/room/room.ts
import { Player } from "../player/player";

export class Room {
    key: string = ""
    roundsQuantity: number
    playersQuantity: number
    playersInTheRoom: number = 1
    currentRound: number = 1
    playerOwner: string
    categoriesNames: string[]
    drawnLetters: string[] = [""]
    players: Player[] = []
    stop = { sent: false, sender: -1 }

    constructor(roundsQuantity: number, playersQuantity: number, playerOwner: string, categoriesNames: string[]) {
        this.roundsQuantity = roundsQuantity
        this.playersQuantity = playersQuantity
        this.playerOwner = playerOwner
        this.categoriesNames = categoriesNames
        this.players.push(new Player(playerOwner, categoriesNames, 0, true))
    }

}
